### PR TITLE
#75

### PR DESCRIPTION
**Pull request by mensfeld**
Related to #75 
@agwozdowski  and @pavlo-vavruk  please review

### CODE REVIEW

**Comment by agwozdowski** [comment]
👍

File: lib/karafka/setup/config.rb [hunk 1]
```diff
@@ -19,20 +28,24 @@ class << self
...
-        :zookeeper_hosts,
-        :kafka_hosts
+        zookeeper: %i( hosts ),
+        kafka: %i( hosts )
       ].freeze
 
-      SETTINGS.each do |attr_name|
+      # We assume that we can have up to two levels of settings (root settings + additional node)
+      # No more nodes are supported (1 level nestings)
+      ROOT_SETTINGS = SETTINGS.map { |set| set.is_a?(Hash) ? set.keys : set }.flatten
```
Comment (pavlo-vavruk) [inline lib/karafka/setup/config.rb]: @mensfeld Use `flat_map` here.

File: lib/karafka/setup/defaults.rb [hunk 2]
```diff
@@ -4,17 +4,19 @@ module Setup
     # use one hidden under a proper method from this class
     class Defaults
       class << self
-        # @return [Array<String>] all Kafka hosts (with ports)
-        # @note If hosts were not provided Karafka will ask Zookeeper for Kafka brokers.
+        # @return [Hash>] all Kafka default settings
```
Comment (pavlo-vavruk) [inline lib/karafka/setup/defaults.rb]: ` [Hash] ` typo

File: lib/karafka/setup/config.rb (hunk 1 above)
Comment (mensfeld) [inline lib/karafka/setup/config.rb]: @pavlo-vavruk updated :)

**Comment by pavlo-vavruk** [comment]
@mensfeld reviewed

**Pull request merged**